javalib/src/main/scala/java/time/temporal/ChronoField.scala

```diff
@@ -0,0 +1,151 @@
+
+  val CLOCK_HOUR_OF_AMPM = new ChronoField("ClockHourOfAmPm", 11,
+    ValueRange.of(1, 12), false, HOURS, HALF_DAYS)
+
+  val HOUR_OF_DAY = new ChronoField("HourOfDay", 12,
+    ValueRange.of(0, 23), false, HOURS, DAYS)
+
+  val CLOCK_HOUR_OF_DAY = new ChronoField("ClockHourOfDay", 13,
+    ValueRange.of(1, 24), false, HOURS, DAYS)
+
+  val AMPM_OF_DAY = new ChronoField("AmPmOfDay", 14,
+    ValueRange.of(0, 1), false, HALF_DAYS, DAYS)
+
+  val DAY_OF_WEEK = new ChronoField("DayOfWeek", 15,
+    ValueRange.of(1, 7), true, DAYS, WEEKS)
+
+  val ALIGNED_DAY_OF_WEEK_IN_MONTH = new ChronoField("AlignedDayOfWeekInMonth",
+    16, ValueRange.of(1, 7), true, DAYS, WEEKS)
+
+  val ALIGNED_DAY_OF_WEEK_IN_YEAR = new ChronoField("AlignedDayOfWeekInYear",
+    17, ValueRange.of(1, 7), true, DAYS, WEEKS)
+
+  val DAY_OF_MONTH = new ChronoField("DayOfMonth", 18,
+    ValueRange.of(1, 28, 31), true, DAYS, MONTHS)
+
+  val DAY_OF_YEAR = new ChronoField("DayOfYear", 19,
+    ValueRange.of(1, 365, 366), true, DAYS, YEARS)
+
+  val EPOCH_DAY = new ChronoField("EpochDay", 20,
+    ValueRange.of(-365249999634l, 365249999634l), true, DAYS, FOREVER)
+
+  val ALIGNED_WEEK_OF_MONTH = new ChronoField("AlignedWeekOfMonth", 21,
+    ValueRange.of(1, 4, 5), true, WEEKS, MONTHS)
+
+  val ALIGNED_WEEK_OF_YEAR = new ChronoField("AlignedWeekOfYear", 22,
+    ValueRange.of(1, 53), true, WEEKS, YEARS)
+
+  val MONTH_OF_YEAR = new ChronoField("MonthOfYear", 23,
+    ValueRange.of(1, 12), true, MONTHS, YEARS)
+
+  val PROLEPTIC_MONTH = new ChronoField("ProlepticMonth", 24,
+    ValueRange.of(-11999999988l, 11999999999l), true, MONTHS, FOREVER)
+
+  val YEAR_OF_ERA = new ChronoField("YearOfEra", 25,
+    ValueRange.of(1, 999999999, 1000000000), true, YEARS, ERAS)
+
+  val YEAR = new ChronoField("Year", 26,
+    ValueRange.of(-999999999, 999999999), true, YEARS, FOREVER)
+
+  val ERA = new ChronoField("Era", 27, ValueRange.of(0, 1), true, ERAS, FOREVER)
+
+  val INSTANT_SECONDS = new ChronoField("InstantSeconds", 28,
+    ValueRange.of(Long.MinValue, Long.MaxValue), false, SECONDS, FOREVER)
+
+  val OFFSET_SECONDS = new ChronoField("OffsetSeconds", 29,
+    ValueRange.of(-64800, 64800), false, SECONDS, FOREVER)
+
+  private val fields = Array(NANO_OF_SECOND, NANO_OF_DAY, MICRO_OF_SECOND,
+    MICRO_OF_DAY, MILLI_OF_SECOND, MILLI_OF_DAY, SECOND_OF_MINUTE,
+    SECOND_OF_DAY, MINUTE_OF_HOUR, MINUTE_OF_DAY, HOUR_OF_AMPM,
+    CLOCK_HOUR_OF_AMPM, HOUR_OF_DAY, CLOCK_HOUR_OF_DAY,
+    AMPM_OF_DAY, DAY_OF_WEEK, ALIGNED_DAY_OF_WEEK_IN_MONTH,
+    ALIGNED_DAY_OF_WEEK_IN_YEAR, DAY_OF_MONTH, DAY_OF_YEAR, EPOCH_DAY,
+    ALIGNED_WEEK_OF_MONTH, ALIGNED_WEEK_OF_YEAR, MONTH_OF_YEAR, PROLEPTIC_MONTH,
+    YEAR_OF_ERA, YEAR, ERA, INSTANT_SECONDS, OFFSET_SECONDS)
+
+  def values(): Array[ChronoField] = fields.clone()
+
+  def valueOf(name: String): ChronoField = {
+    fields.find(name == _.name.toScreamingSnakeCase) match {
+      case Some(field) => field
```

Comment:
Don't use pattern matching on `Option`. Use `fold` instead.
https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#option-and-jsundefor